chatkit-core/src/main/kotlin/com/pusher/chatkit/model/mappers/Room.kt

```diff
@@ -0,0 +1,35 @@
+package com.pusher.chatkit.model.mappers
+
+import com.pusher.chatkit.model.network.ReadStateApiType
+import com.pusher.chatkit.model.network.RoomApiType
+import com.pusher.chatkit.model.network.RoomMembershipApiType
+import com.pusher.chatkit.rooms.Room
+
+internal fun mapToRoom(
+        room: RoomApiType,
+        memberships: RoomMembershipApiType?,
+        readState: ReadStateApiType?
+) = mapToRoom(
+        room = room,
+        memberships = memberships?.userIds.orEmpty().toSet(),
+        unreadCount = readState?.unreadCount
+)
+
+internal fun mapToRoom(
```

Comment:
[Likewise](https://github.com/pusher/chatkit-android/pull/144/files#r366860798), better packaging: `com.pusher.chatkit.rooms.internal.JoinedRoomInternalMapper`.

There should be also `NonJoinedRoomApilMapper` for which we would explicitly see that `membership` and `read_status` are not there.